Cargo.toml

```diff
@@ -83,53 +83,53 @@ base64 = "0.13"
 encoding_rs = "0.8"
 futures-core = { version = "0.3.0", default-features = false }
 futures-util = { version = "0.3.0", default-features = false }
-http-body = "0.3.0"
-hyper = { version = "0.13.4", default-features = false, features = ["tcp"] }
+http-body = "0.4.0"
+hyper = { version = "0.14", default-features = false, features = ["tcp", "http1", "http2", "client"] }
 lazy_static = "1.4"
 log = "0.4"
 mime = "0.3.7"
 percent-encoding = "2.1"
-tokio = { version = "0.2.5", default-features = false, features = ["tcp", "time"] }
+tokio = { version = "1.0", default-features = false, features = ["net", "time"] }
 pin-project-lite = "0.2.0"
 ipnet = "2.3"
 
 # Optional deps...
 
 ## default-tls
-hyper-tls = { version = "0.4", optional = true }
+hyper-tls = { git = "https://github.com/teozkr/hyper-tls.git", branch = "chore/tokio-1.0", optional = true }
 native-tls-crate = { version = "0.2", optional = true, package = "native-tls" }
-tokio-tls = { version = "0.3.0", optional = true }
+tokio-native-tls = { version = "0.3.0", optional = true }
 
 # rustls-tls
-hyper-rustls = { version = "0.21", default-features = false, optional = true }
-rustls = { version = "0.18", features = ["dangerous_configuration"], optional = true }
-tokio-rustls = { version = "0.14", optional = true }
-webpki-roots = { version = "0.20", optional = true }
-rustls-native-certs = { version = "0.4", optional = true }
+hyper-rustls = { version = "0.22.1", default-features = false, optional = true }
+rustls = { version = "0.19", features = ["dangerous_configuration"], optional = true }
+tokio-rustls = { version = "0.22", optional = true }
+webpki-roots = { version = "0.21", optional = true }
+rustls-native-certs = { version = "0.5", optional = true }
 
 ## cookies
 cookie_crate = { version = "0.14", package = "cookie", optional = true }
 cookie_store = { version = "0.12", optional = true }
 time = { version = "0.2.11", optional = true }
 
 ## compression
-async-compression = { version = "0.3.0", default-features = false, features = ["stream"], optional = true }
-
+async-compression = { version = "0.3.7", default-features = false, features = ["tokio"], optional = true }
+tokio-util = { version = "0.6.0", default-features = false, features = ["codec", "io"], optional = true }
 
 ## socks
-tokio-socks = { version = "0.3", optional = true }
+tokio-socks = { version = "0.5", optional = true }
 
 ## trust-dns
-trust-dns-resolver = { version = "0.19", optional = true }
+trust-dns-resolver = { git = "https://github.com/bluejekyll/trust-dns.git", branch = "main", optional = true }
```

Comment:
I've wondered about removing the explicit trust-dns support, and having people use it via what's proposed in #1125. What do y'all think?

Reply:
I think that would make sense, and might also actually provide opportunities to do closer integration. On the trust-dns side I've seen requests for some way to share configuration between a resolver used directly and the resolver used by reqwest.

Reply:
FYI, release for Trust-DNS is currently publishing.

Reply:
I think that you should transition the `trust-dns` feature to use the API described in #1125, mark it as deprecated and later remove it. The first step will help test and document how to migrate.

Reply:
Maybe @bluejekyll can weigh in on #1125 and that trait could be published in a separate crate.